xmpp.go

```diff
@@ -570,8 +570,11 @@ func (c *Client) init(o *Options) error {
 	}
 
 	// If the server requires STARTTLS, attempt to do so.
-	if f, err = c.startTLSIfRequired(f, o, domain); err != nil {
-		return err
+	// unless the client has specified NoTLS
+	if !o.NoTLS {
```

Comment:
I am not sure if we should use the directTLS setting to control whether StartTLS is used. I'll check if there is an alternative.